packages/lesswrong/server/embeddings.ts

```diff
@@ -0,0 +1,84 @@
+import Posts from "../lib/collections/posts/collection";
+import { PostEmbeddingsRepo } from "./repos";
+import { forEachDocumentBatchInCollection } from "./manualMigrations/migrationUtils";
+import { getOpenAI } from "./languageModels/languageModelIntegration";
+import { htmlToText } from "html-to-text";
+import { Globals } from "./vulcan-lib";
+import { inspect } from "util";
+import md5 from "md5";
+
+export const DEFAULT_EMBEDDINGS_MODEL = "text-embedding-ada-002";
+
+type EmbeddingsResult = {
+  embeddings: number[],
+  model: string,
+}
+
+const getEmbeddingsFromApi = async (text: string): Promise<EmbeddingsResult> => {
+  const api = await getOpenAI();
+  if (!api) {
+    throw new Error("OpenAI client is not configured");
+  }
+  const model = DEFAULT_EMBEDDINGS_MODEL;
+  const result = await api.createEmbedding({
+    input: text,
+    model,
+  });
+  const embeddings = result?.data?.data?.[0].embedding;
+  if (
+    !embeddings ||
+    !Array.isArray(embeddings) ||
+    !embeddings.length ||
+    typeof embeddings[0] !== "number"
+  ) {
+    throw new Error(`Invalid API response: ${inspect(result, {depth: null})}`);
```

Comment:
I would truncate that inspect, it might be quite long?

Reply:
It should be an edge case - if an error happens then I want as much info as I can get.